src/java.desktop/share/classes/javax/swing/JEditorPane.java

```diff
@@ -473,68 +473,76 @@ public void setPage(URL page) throws IOException {
             throw new IOException("invalid url");
         }
         URL loaded = getPage();
-
+        Object postData = getPostData();
 
         // reset scrollbar
         if (!page.equals(loaded) && page.getRef() == null) {
-            scrollRectToVisible(new Rectangle(0,0,1,1));
+            scrollRectToVisible(new Rectangle(0, 0, 1, 1));
         }
```

Comment:
This should go below the `if`-block:

```java
if ((postData == null) && (page.sameFile(loaded))) {
```

And the condition becomes redundant. After that `if`-block, we know for sure the page is going to be reloaded. That is the code should look like this:

```java
        if ((postData == null) && (page.sameFile(loaded))) {
            // The same page with different reference
            /* <the code you have> */
            return;
        }

        // different url or POST method, load the new content

        // reset scrollbar
        scrollRectToVisible(new Rectangle(0, 0, 1, 1));
```